src/careamics/config/data/ng_data_model.py

```diff
@@ -0,0 +1,373 @@
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from pprint import pformat
+from typing import Annotated, Any, Literal, Optional, Union
+from warnings import warn
+
+import numpy as np
+from numpy.typing import NDArray
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    Field,
+    PlainSerializer,
+    field_validator,
+    model_validator,
+)
+from typing_extensions import Self
+
+from ..transformations import XYFlipModel, XYRandomRotate90Model
+from ..validators import check_axes_validity
+from .patching_strategies import (
+    RandomPatchingModel,
+    SequentialPatchingModel,
+    TiledPatchingModel,
+    WholePatchingModel,
+)
+
+# TODO: Validate the specific sizes of tiles and overlaps given UNet constraints
+#   - needs to be done in the Configuration
+#   - patches and overlaps sizes must also be checked against dimensionality
+
+
+# TODO: validation axes vs patch sizes
+# TODO: is 3D updated anywhere in the code?
+# TODO: can we tell when it is made for training but has the wrong patching?
+#       - or is the responsibility of the creator?
+
+
+def np_float_to_scientific_str(x: float) -> str:
+    """Return a string scientific representation of a float.
+
+    In particular, this method is used to serialize floats to strings, allowing
+    numpy.float32 to be passed in the Pydantic model and written to a yaml file as str.
+
+    Parameters
+    ----------
+    x : float
+        Input value.
+
+    Returns
+    -------
+    str
+        Scientific string representation of the input value.
+    """
+    return np.format_float_scientific(x, precision=7)
+
+
+Float = Annotated[float, PlainSerializer(np_float_to_scientific_str, return_type=str)]
+"""Annotated float type, used to serialize floats to strings."""
+
+PatchingStrategies = Union[
+    RandomPatchingModel,
+    SequentialPatchingModel,
+    TiledPatchingModel,
+    WholePatchingModel,
+]
+"""Patching strategies."""
+
+
+class NGDataConfig(BaseModel):
+    """Next-Generation Dataset configuration.
+
+    If std is specified, mean must be specified as well. Note that setting the std first
+    and then the mean (if they were both `None` before) will raise a validation error.
+    Prefer instead `set_mean_and_std` to set both at once. Means and stds are expected
+    to be lists of floats, one for each channel. For supervised tasks, the mean and std
+    of the target could be different from the input data.
+
+    All supported transforms are defined in the SupportedTransform enum.
+    """
+
+    # Pydantic class configuration
+    model_config = ConfigDict(
+        validate_assignment=True,
+    )
+
+    # Dataset configuration
+    # TODO what becomes of custom? is custom passing imagestacks?
+    data_type: Literal["array", "tiff", "zarr", "custom"]
+    """Type of input data."""
+
+    axes: str
+    """Axes of the data, as defined in SupportedAxes."""
+
+    patching: Union[PatchingStrategies] = Field(..., discriminator="name")
```

Comment:
`PatchingStrategies` is already a Union

```suggestion
    patching: PatchingStrategies = Field(..., discriminator="name")
```

Reply:
Good catch, thanks!